src/savegame_json.cpp

```diff
@@ -517,14 +517,13 @@ void Character::load( const JsonObject &data )
         data.read( "mutations", my_mutations );
     }
     for( auto it = my_mutations.begin(); it != my_mutations.end(); ) {
-        const auto &mid = it->first;
+        const trait_id &mid = it->first;
         if( mid.is_valid() ) {
             on_mutation_gain( mid );
             cached_mutations.push_back( &mid.obj() );
```

Comment:
You need this again, or it's an infinite loop.
```suggestion
            cached_mutations.push_back( &mid.obj() );
            ++it;
```